plugins/inputs/chrony/chrony.go

```diff
@@ -5,125 +5,179 @@ import (
 	_ "embed"
 	"errors"
 	"fmt"
-	"os/exec"
+	"net"
+	"net/url"
+	"os"
 	"strconv"
 	"strings"
 	"time"
 
+	fbchrony "github.com/facebook/time/ntp/chrony"
+
 	"github.com/influxdata/telegraf"
-	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/config"
 	"github.com/influxdata/telegraf/plugins/inputs"
 )
 
 //go:embed sample.conf
 var sampleConfig string
 
-var (
-	execCommand = exec.Command // execCommand is used to mock commands in tests.
-)
-
 type Chrony struct {
-	DNSLookup bool `toml:"dns_lookup"`
-	path      string
+	Server    string          `toml:"server"`
+	Timeout   config.Duration `toml:"timeout"`
+	DNSLookup bool            `toml:"dns_lookup"`
+	Log       telegraf.Logger `toml:"-"`
+
+	conn   net.Conn
+	client *fbchrony.Client
 }
 
 func (*Chrony) SampleConfig() string {
 	return sampleConfig
 }
 
 func (c *Chrony) Init() error {
-	var err error
-	c.path, err = exec.LookPath("chronyc")
-	if err != nil {
-		return errors.New("chronyc not found: verify that chrony is installed and that chronyc is in your PATH")
+	if c.Server != "" {
+		// Check the specified server address
+		u, err := url.Parse(c.Server)
+		if err != nil {
+			return fmt.Errorf("parsing server address failed: %w", err)
+		}
+		switch u.Scheme {
+		case "unix":
+			// Keep the server unmodified
+		case "udp":
+			// Check if we do have a port and add the default port if we don't
+			if u.Port() == "" {
+				u.Host += ":323"
+			}
+			// We cannot have path elements in an UDP address
+			if u.Path != "" {
+				return fmt.Errorf("path detected in UDP address %q", c.Server)
+			}
+			u = &url.URL{Scheme: "udp", Host: u.Host}
+		case "":
```

Comment:
I'd rather you fall the default case here and require the scheme for the user to be explicit about a scheme. 

Are you trying to parse values like `server:323`? If so, url.Parse, will report the scheme as 'server'. The only way to have an empty string for the scheme is if the user uses `//server:385`.

Reply:
Nice catch. Will change to require an explicit scheme.